src/main/scala/org/openeo/opensearch/backends/OscarsClient.scala

```diff
@@ -20,6 +20,7 @@ class OscarsClient(val endpoint: URL, val isUTM:Boolean = false) extends OpenSea
         .param("sortKeys", key)
         .param("count", "1")
         .params(attributeValues.mapValues(_.toString).toSeq)
+        .timeout(connTimeoutMs = 10000, readTimeoutMs = 40000)
```

Comment:
I would delegate to superclass method org.openeo.opensearch.OpenSearchClient#http instead.

Reply:
Good idea. As a bonus, redirects can be followed too here.